### PR TITLE
Fix Gorm migrate issue

### PR DESCRIPTION
**Pull request by qingxinhome**
## What type of PR is this?

- [ ] API-change
- [x] BUG
- [ ] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #11527
issue #11522

## What this PR does / why we need it:
Solving the gorm migration problem of mo

### CODE REVIEW

**Comment by matrix-meow** [comment]
@qingxinhome Thanks for your contributions!

## Review:

### Title
The title of the pull request is "Fix Gorm migrate issue". The title is concise and provides a clear indication of the purpose of the pull request.

### Body
The body of the pull request includes the following sections:
- "What type of PR is this?" - This section provides checkboxes to indicate the type of pull request. The checkbox for "BUG" is checked, indicating that this is a bug fix.
- "Which issue(s) this PR fixes" - This section lists the issues that this pull request fixes, specifically issue #11527 and issue #11522.
- "What this PR does / why we need it" - This section briefly explains that the pull request solves a Gorm migration problem.

The body of the pull request provides relevant information about the purpose and context of the changes.

### Changes
The changes made in the pull request include modifications to the `mysql_protocol.go`, `session.go`, and `visit_plan.go` files.

In `mysql_protocol.go`, the `authenticateUser` function is modified to include additional parameters: `dbName`, `authResponse`, `salt`, and `checkPassword`. The `authenticateDBName` function is added to authenticate the database name. The commented code block that checks the database name is not removed.

In `session.go`, the `AuthenticateUser` function is modified to include the `dbName` parameter and to authenticate the database name if it is provided. The `AuthenticateDBName` function is added to authenticate the database name.

In `visit_plan.go`, the `exploreNode` function is modified to apply the `rule` to the `OrderBy` expressions.

### Problems
1. Commented code block: In `mysql_protocol.go`, there is a commented code block that checks the database name. This code block should either be removed or uncommented if it is necessary.
2. Unused variables: In `session.go`, the `AuthenticateUser` function includes a variable `txnMeta` that is not used. This variable should be removed to avoid confusion.
3. Lack of comments: The changes made in `session.go` and `visit_plan.go` lack comments explaining the purpose and functionality of the added or modified code. Adding comments would improve the readability and maintainability of the code.

### Suggestions
1. Remove or uncomment the commented code block in `mysql_protocol.go` if it is necessary.
2. Remove the unused variable `txnMeta` in `session.go`.
3. Add comments to explain the purpose and functionality of the added or modified code in `session.go` and `visit_plan.go`.

### Optimization
No specific optimization suggestions are needed for this pull request.

**Review by ouyuanning — APPROVED** [review]

File: pkg/frontend/mysql_protocol.go [hunk 1]
```diff
@@ -1271,6 +1276,20 @@ func (mp *MysqlProtocolImpl) Authenticate(ctx context.Context) error {
 		return err
 	}
 
+	//if dbName := mp.GetDatabaseName(); dbName != "" {
```
Comment (daviszhen) [inline pkg/frontend/mysql_protocol.go:1279]: 不要就删了吧

File: pkg/frontend/session.go [hunk 2]
```diff
@@ -1503,13 +1503,54 @@ func (ses *Session) AuthenticateUser(userInput string) ([]byte, error) {
...
+	}
+	//------------------------------------------------------------------------------------------------------------------
 	// record the id :routine pair in RoutineManager
 	ses.getRoutineManager().accountRoutine.recordRountine(tenantID, ses.getRoutine(), accountVersion)
 	logInfo(ses, sessionInfo, tenant.String())
 
 	return GetPassWord(pwd)
 }
 
+func (ses *Session) AuthenticateDBName(dbName string) error {
```
Comment (daviszhen) [inline pkg/frontend/session.go:1536]: 这个也是不用的吧。也删了吧。

File: pkg/frontend/mysql_protocol.go [hunk 3]
```diff
@@ -1194,6 +1194,11 @@ func (mp *MysqlProtocolImpl) authenticateUser(ctx context.Context, authResponse
 	return nil
 }
 
+func (mp *MysqlProtocolImpl) authenticateDBName(ctx context.Context, authResponse []byte) error {
```
Comment (daviszhen) [inline pkg/frontend/mysql_protocol.go:1197]: 这个也删了吧。

**Review by daviszhen — CHANGES_REQUESTED** [review]

File: pkg/frontend/mysql_protocol.go (hunk 1 above)
Comment (qingxinhome) [reply pkg/frontend/mysql_protocol.go:1279]: 好的

File: pkg/frontend/session.go (hunk 2 above)
Comment (qingxinhome) [reply pkg/frontend/session.go:1536]: 好的

File: pkg/frontend/mysql_protocol.go (hunk 3 above)
Comment (qingxinhome) [reply pkg/frontend/mysql_protocol.go:1197]: 好的

**Review by daviszhen — APPROVED** [review]

**Review by heni02 — APPROVED** [review]

**Review by aressu1985 — APPROVED** [review]

**Pull request merged**